services/src/main/scala/cromwell/services/metadata/impl/DeleteMetadataActor.scala

```diff
@@ -0,0 +1,30 @@
+package cromwell.services.metadata.impl
+
+import akka.actor.{Actor, ActorLogging}
+import cromwell.services.MetadataServicesStore
+import cromwell.services.metadata.MetadataService.{DeleteMetadataAction, DeleteMetadataFailedResponse, DeleteMetadataSuccessfulResponse}
+
+import scala.util.{Failure, Success}
+
+class DeleteMetadataActor extends Actor
+  with ActorLogging
+  with MetadataDatabaseAccess
+  with MetadataServicesStore {
+
+  implicit val ec = context.dispatcher
+
+  override def receive: Receive = {
+    case action@DeleteMetadataAction(workflowId, replyTo, maxAttempts) =>
+      val deleteFromDbFuture = deleteNonLabelMetadataEntriesForWorkflow(workflowId)
+      deleteFromDbFuture onComplete {
+        case Success(_) => replyTo ! DeleteMetadataSuccessfulResponse(workflowId)
+        case Failure(ex) if maxAttempts > 0 =>
```

Comment:
Won't this make up to two attempts if `maxAttempts` is 1?  Perhaps `maxRetries` would be more accurate.

Reply:
it looks like `maxAttempts` is in the existing "puts" code and does actually represent the maximum number of attempts